🌐 Update Korean translation for `docs/ko/docs/virtual-environments.md`

sungchan1 opened the pull request:
Hello! 👋 This is my first contribution to FastAPI Korean translation 😊

I translated the `docs/ko/docs/virtual-environments.md` file into Korean.
Since it's a core topic in Python projects, I noticed it hasn't been localized yet, so I decided to help!

I tried to stay as close as possible to the original English format and meaning.

If there is any feedback, I’d be happy to update it right away.

감사합니다!

<img width="1913" alt="image" src="https://github.com/user-attachments/assets/6de77990-2ad9-4d76-929c-067f67373939" />



Before : https://fastapi.tiangolo.com/ko/virtual-environments/
After.   : https://2d5c56b3.fastapitiangolo.pages.dev/ko/virtual-environments/

sungchan1 committed:
Update docs/ko/docs/virtual-environments.md

sungchan1 committed:
Update docs/ko/docs/virtual-environments.md

pre-commit-ci[bot] committed:
🎨 [pre-commit.ci] Auto format from pre-commit.com hooks

sungchan1 committed:
Merge branch 'master' into fix/ko-typo-virtual-environments

sodaMelon commented:
hi @sungchan1
just saw your email asking me to review this PR
so I'll comment later (maybe...... in this evening!!!!) 

안녕하세요 @sungchan1 님
이메일로 리뷰 부탁하신 것 방금확인했어요.
이따가 코멘트 드리겠습니다(오늘 저녁 예정!! 평일엔 제가 조금 바쁘네요 ㅠㅠ)

sodaMelon commented on docs/ko/docs/virtual-environments.md:
```diff
@@ -0,0 +1,854 @@
+
+/// tip
+
+프로젝트를 작업할 때마다, **터미널을 새로 열었을 때마다** 이 작업을 꼭 해주세요.
```
```suggestion
**터미널을 새로 열고** 프로젝트 작업을 시작할 때는, **항상 이 작업을** 해주세요.
```
~마다,~마다가 
두번 나와서 좀 어색해보이길래... 문장 추천드립니다.

sodaMelon commented on docs/ko/docs/virtual-environments.md:
```diff
@@ -0,0 +1,854 @@
+
+/// tip
+
+여전히 **패키지 이름과 버전 정보를 파일에 정리해두는 것**(예: `requirements.txt` 또는 `pyproject.toml`)이 **(매우) 좋은 습관**입니다.
```
```suggestion
**패키지 이름과 버전 정보를 파일에 정리해두는 것**(예: `requirements.txt` 또는 `pyproject.toml`)은 **(매우) 좋은 습관**입니다.
```

sodaMelon commented on docs/ko/docs/virtual-environments.md:
```diff
@@ -0,0 +1,854 @@
+
+</div>
+
+이렇게 하면 이후에 `python` 명령어를 실행해도, 가상 환경의 Python이나 그 안에 설치된 패키지들을 사용하지 않게 됩니다.
```
```suggestion
이렇게 하면 이후에 `python` 명령어를 실행했을 때, 가상 환경의 Python이나 그 안에 설치된 패키지들을 사용하지 않게 됩니다.
```

sodaMelon commented on docs/ko/docs/virtual-environments.md:
```diff
@@ -0,0 +1,854 @@
+
+그렇다면, 전역 Python 환경에 패키지를 설치하는 것의 문제는 무엇일까요?
+
+어느 시점이 되면, 여러분은 아마 **서로 다른 패키지들**에 의존하는 여러 개의 프로그램을 작성하게 될 것입니다. 그리고 이들 중 일부는 **같은 패키지의 서로 다른 버전**을 필요로 할 수 있습니다. 😱
```
```suggestion
어느 시점이 되면, **서로 다른 패키지들**에 의존하는 여러 개의 프로그램을 작성하게 될 것입니다. 그리고 이들 중 일부는 **같은 패키지의 서로 다른 버전**을 필요로 할 수 있습니다. 😱
```

sodaMelon commented on docs/ko/docs/virtual-environments.md:
````diff
@@ -0,0 +1,854 @@
+예를 들어, `philosophers-stone`이라는 프로젝트를 만들었다고 가정해봅시다. 이 프로그램은 `harry`라는 패키지의 **버전 1**에 의존합니다. 따라서 `harry`를 설치해야 합니다.
+
+
+```mermaid
````
## en
![image](https://github.com/user-attachments/assets/95382d12-e2f9-433a-a4d7-28a38e62d60e)

## ko
![image](https://github.com/user-attachments/assets/1e460eaa-1cc1-4626-8744-0bff37ef30fa)

flow chart부분 문자 간격이 좀 다르네요? 🤔
원인은 잘모르겠지만..

sodaMelon reviewed (commented):
텍스트 양이 많아서 일단 여기서 한번 끊을게요!

sodaMelon commented on docs/ko/docs/virtual-environments.md:
````diff
@@ -0,0 +1,854 @@
+    stone(philosophers-stone) -->|requires| harry-1[harry v1]
+```
+
+그런데 나중에 prisoner-of-azkaban이라는 또 다른 프로젝트를 만들게 되었고, 이 프로젝트도 harry 패키지에 의존하지만 버전 3이 필요합니다.
````
```suggestion
그런데 나중에 `prisoner-of-azkaban`이라는 또 다른 프로젝트를 만들게 되었고, 이 프로젝트도 `harry` 패키지에 의존합니다. 그런데 이 프로젝트는 **`harry`**의  **버전  `3`**이 필요합니다.
```

sodaMelon commented on docs/ko/docs/virtual-environments.md:
````diff
@@ -0,0 +1,854 @@
+    azkaban(prisoner-of-azkaban) --> |requires| harry-3[harry v3]
+```
+
+이제 문제가 발생합니다. 전역 환경에 패키지를 설치하게 되면, 어떤 버전의 harry를 설치할지를 선택해야 하기 때문입니다.
````
```suggestion
이제 문제가 발생합니다. (로컬 가상환경 대신) 전역 환경에 패키지를 설치하게 되면, 어떤 버전의 harry를 설치할지를 선택해야 하기 때문입니다.
```

sodaMelon commented on docs/ko/docs/virtual-environments.md:
```diff
@@ -0,0 +1,854 @@
+모든 패키지를 전역 환경에 설치하면서 발생하는 문제를 해결하는 방법은
+**프로젝트마다 가상 환경을 따로 사용하는 것**입니다.
+
+가상 환경은 하나의 **디렉터리**이며, 전역 환경과 유사한 구조를 가지고 있습니다.
+여기에서 각 프로젝트에 필요한 패키지를 독립적으로 설치할 수 있습니다.
```
이 문장은 원문보다 의미가 더 잘 드러난 것 같아요. 훌륭합니다.

sodaMelon commented on docs/ko/docs/virtual-environments.md:
````diff
@@ -0,0 +1,854 @@
+C:\Windows\System32
+```
+
+That means that the system would look for programs in:
````
```suggestion
That means that the system would look for programs in:
```
여기는 번역이 안되어있네요?

sodaMelon commented on docs/ko/docs/virtual-environments.md:
```diff
@@ -0,0 +1,854 @@
+
+/// tip
+
+하나의 가상 환경을 활성화한 상태에서 **다른 프로젝트로 이동**하면 문제가 발생할 수 있습니다.
```
```suggestion
하나의 가상 환경을 활성화한 상태에서 **다른 프로젝트로 이동**하면 문제가 발생할 수 있습니다.
```
여기도 원본 텍스트랑 의미가 좀 달라져서... 코멘트 드려요.
나중에 원문이 바뀌면 따라가기 힘들기때문에.....한번 고민 부탁드립니다.

sungchan1 replied:
원문이 바뀌는걸 생각하지 못했네요. 전체적으로 직역에 가깝게 바꾸겠습니다.

sodaMelon commented on docs/ko/docs/virtual-environments.md:
```diff
@@ -0,0 +1,854 @@
+
+예를 들어 `philosophers-stone`이라는 프로젝트에서 작업 중이라고 해보겠습니다. 이때 해당 가상 환경을 **활성화**하고, 필요한 패키지를 설치하며 작업을 진행합니다.
+
+그런데 이제는 **다른 프로젝트**인 `prisoner-of-azkaban`으로 작업을 옮기고 싶어졌다고 해볼게요.
```
```suggestion
그런데 이제는 **다른 프로젝트**인 `prisoner-of-azkaban`을 작업하고 싶어졌습니다.
```

sungchan1 replied:
혹시 한번에 의미가 오지 않을까봐, 해리포터 부분 번역 추가했습니다.

sodaMelon commented on docs/ko/docs/virtual-environments.md:
```diff
@@ -0,0 +1,854 @@
+
+## 대안들
+
+이 문서는 여러분이 Python 프로젝트를 시작하고, 그 내부에서 어떤 일이 일어나는지를 **직접 이해할 수 있도록 도와주는 간단한 가이드**입니다.
```
```suggestion
이 문서는 여러분이 Python 프로젝트를 시작하고, *그 내부에서* 어떻게 돌아가는지 알려주는 간단한 가이드입니다.
```

sodaMelon commented on docs/ko/docs/virtual-environments.md:
```diff
@@ -0,0 +1,854 @@
+* 각 프로젝트에 대해 **가상 환경을 자동으로 관리**합니다
+* **패키지를 설치**할 수 있습니다
+* 프로젝트의 **의존성과 버전**을 관리합니다
+* 설치된 패키지들과 그 버전을 **정확히 고정(lock)**해서,
+  개발 중인 환경과 **운영 환경**이 완전히 동일하게 작동할 수 있도록 보장합니다
```
```suggestion
* 설치된 패키지들과 그 버전을 **정확히 고정(lock)**해서,
  개발 중인 환경과 **운영 환경**이 완전히 동일하게 작동할 수 있도록 보장합니다
```
이 문장도 한국어 상으로는 완벽하지만 의미를 조금더 생각해보셨으면 좋겠습니다!

sungchan1 replied:
"정확히 고정(lock) " 이 부분이 어렵네요.. 일단 그대로 두고 

개발 중인 환경과 **운영 환경**이 완전히 동일하게 작동할 수 있도록 보장합니다
-> 개발 환경과 운영 환경이 완전히 동일하게 작동할 수 있도록 보장

sodaMelon commented on docs/ko/docs/virtual-environments.md:
```diff
@@ -0,0 +1,854 @@
+
+여기까지 모두 읽고 이해했다면, 이제 여러분은 많은 개발자들보다 **훨씬 더 깊이 있게 가상 환경을 이해하고 있는 사람**입니다. 🤓
+
+이러한 세부적인 내용을 알고 있는 것은, 언젠가 복잡해 보이는 문제를 디버깅할 때 분명히 큰 도움이 될 것입니다. 왜냐하면 이제는 여러분이 **이 모든 것들이 내부에서 어떻게 작동하는지** 알고 있기 때문입니다. 😎
```
```suggestion
이러한 세부적인 내용을 알고 있는 것은, 언젠가 복잡해 보이는 문제를 디버깅할 때 분명히 큰 도움이 될 것입니다. 이제는 **이 모든 것들이 내부에서 어떻게 작동하는지** 알고 있기 때문입니다. 😎
```

sodaMelon commented:
@sungchan1 finally i finished review. 
there are too many comments.... but...maybe you can check all😅 (bc this doc is longer than others)
리뷰 마쳤습니다.
코멘트가 너무 많지만, 확인 잘부탁드립니다.😅 (이 문서 자체가 다른 문서보다 기네요..)

+) 전체적으로 문장이 자연스럽습니다. 잘해주셨어요! 단, 자연스럽게 번역하다보니 원문과 의미가 좀 멀어질 수 있는 것들을 체크해두었으니......... 현실에서 소리내어서 읽어보시고 자연스러운 쪽을 선택하세요.

sodaMelon commented on docs/ko/docs/virtual-environments.md:
```diff
@@ -0,0 +1,854 @@
+// 이제 python을 실행하면, 이 가상 환경에 설치된 sirius 패키지를 찾게 됩니다 ✨
+$ python main.py
+
+I solemnly swear 🐺
```
@sungchan1 이 문서에서는 예시를 해리포터 1권 3권 얘기를 예시로 들고있는데요. 
혹시 이 문장도 이해 하셨나요? 🐺<- 이 emoji의 의미하구요

sungchan1 replied:
아 이 부분 이해못했네요...
다시 검토하면서 수정하겠습니다.

sodaMelon replied:
저도 이거는 사실 번역을 해야할까 고민하긴했어요 context문제라 🤔ㅋㅋㅋㅋ 생각해보고 괜찮은 방법이 생각나면, 더 코멘트 드릴게요 !!

sungchan1 replied:
머글 입장에서 최선을 다해 노력했습니다..

못된 짓을 꾸미고 있음을 엄숙히 맹세합니다.🧙
ImportError는 이제 없습니다. 🐺

sodaMelon replied:
ㅋㅋㅋㅋ 고생하셨어요~ 한국은 영어 문화권이 아니다보니, 설명이 필요했는데 이정도면 충분한 것 같습니다

sungchan1 committed:
Update docs/ko/docs/virtual-environments.md

- 읽었을 때, 어색한 단어, 조사 수정했습니다.
- Python에 기본 포함된 -> Python 표준 라이브러리 통일했습니다.
- 잘못 처리된 굵은 단어, `` 단어 수정했습니다.
- 소제목 통일 (~~하기 -> ~~)
  - 패키지 설치하기 -> 패키지 설치
  - `requirements.txt`에서 설치하기 -> `requirements.txt`에서 설치
- 해리 포터 예시 해석 추가
  - philosophers-stone -> 마법사의 돌(philosophers-stone)
  - prisoner-of-azkaban -> 아즈카반의 죄수(prisoner-of-azkaban)

pre-commit-ci[bot] committed:
🎨 [pre-commit.ci] Auto format from pre-commit.com hooks

sungchan1 commented on docs/ko/docs/virtual-environments.md:
```diff
@@ -0,0 +1,854 @@
+**프로젝트마다 가상 환경을 따로 사용하는 것**입니다.
+
+가상 환경은 하나의 **디렉터리**이며, 전역 환경과 유사한 구조를 가지고 있습니다.
+여기에서 각 프로젝트에 필요한 패키지를 독립적으로 설치할 수 있습니다.
```
직역에 가깝게 변경했습니다!

sungchan1 committed:
Update docs/ko/docs/virtual-environments.md

- 오타 수정했습니다
  동일하장 작동할 수 있도록 보 -> 동일하게 작동할 수 있도록 보장

sodaMelon commented on docs/ko/docs/virtual-environments.md:
```diff
@@ -0,0 +1,846 @@
+
+## 대안들
+
+이 문서는 여러분이 Python 프로젝트를 시작하고, *그 내부에서* 어떻게 돌아가는지 알려주는 간단한 가이드입니다.
```
```suggestion
이 문서는 여러분이 Python 프로젝트를 시작하고, **그 내부에서** 어떻게 돌아가는지 알려주는 간단한 가이드입니다.
```
my mistake..😅 
* -> **로 바꿨어요  (italic -> bold)

sungchan1 replied:
감사합니다!!
문서가 길어서 지쳤는데, 하나하나 읽고 코멘트 달아주셔서 힘을 얻었습니다.
처음에는 뜻 전달 우선이었는데, 조언 주셔서 읽을 때 매끄러움을 생각할 수 있었어요!

sodaMelon reviewed (changes requested):
LGTM!! 
i left just 1 request comment for my mistake. 😅
if you modify it, and I'll approve!

잘해주셨습니다~~ 
제가 실수한 거에 대해서 코멘트 남겨뒀어요! ㅋㅋ ㅠㅠㅠ 
그거만 반영해주시면 approve 드릴게요~~ 고생하셨습니다!!!

sungchan1 committed:
Update docs/ko/docs/virtual-environments.md

Co-authored-by: sodam <66295123+sodaMelon@users.noreply.github.com>

sungchan1 committed:
Merge branch 'master' into fix/ko-typo-virtual-environments

Zerohertz reviewed (changes requested):
```markdown
/// warning | 경고
```

위와 같이 전체적으로 `| {한국어}` 주석이 필요할 것으로 보입니다 ㅎㅎ!

그 외에는 잘 번역해 주셔서 감사드립니다 ㅎㅎ
위 사항 수정 후 approve 하겠습니다.

sungchan1 committed:
📝 Update virtual-environments.md for improved clarity and consistency in tips and information sections

sungchan1 commented on docs/ko/docs/virtual-environments.md:
```diff
@@ -0,0 +1,846 @@
+
+///
+
+/// tip
```
수정사항 반영했는데, 바로 approve 주셔서 감사합니다! 😊😊

sungchan1 committed:
Merge branch 'master' into fix/ko-typo-virtual-environments

sungchan1 committed:
Merge branch 'master' into fix/ko-typo-virtual-environments

sungchan1 committed:
Merge branch 'master' into fix/ko-typo-virtual-environments
